services/application.js

```diff
@@ -209,7 +206,7 @@ class Application {
         const deletedOne = await this.getApplicationById(document._id);
         let result = null;
         if (deletedOne && await this.dbService.deleteOne(APPLICATION, {_id: document._id})) {
-            result = deletedOne[0];
```

Comment:
Is this change from **deleteOne[0]** to **deleteOne** intentional?

Also, we should NOT delete any application records from DB, should only change status to "Deleted" instead.

Reply:
It was on our to-do list. I have now pushed it for an update.

Reply:
deleteApplication function needs permission control, please add a TODO comment in the function, so we won't forget about it.